### PR TITLE
[Feature] Add hbase engine to linkis

### PR DESCRIPTION
**Pull request by CCweixiao**
### What is the purpose of the change

Add hbase engine to linkis, The ruby command of hbase-shell can be executed through linkis.

### Related issues

Related issues: #2533 

### Brief change log

- Add hbase engine to linkis

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

**Comment by peacewong** [comment]
ping~ @jackxu2011 @jacktao007

File: linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+  public Connection getConnection(Properties prop) {
+    Map<String, String> propMap = getPropMap(prop);
+    return getConnection(propMap);
+  }
+
+  public Connection getConnection(Configuration configuration) {
+    String clusterConnUniqueKey = generateUniqueConnectionKey(configuration);
+    LOG.info("Start to get connection for cluster {}.", clusterConnUniqueKey);
+    try {
+      lock.lock();
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java:107]: Is it better to first determine whether it contains the corresponding key and then lock it?

File: linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+  }
+
+  public Connection getConnection(Map<String, String> prop) {
+    Configuration configuration = buildConfiguration(prop);
+    return getConnection(configuration);
+  }
+
+  private void doKerberosLogin(Configuration configuration) {
+    String principal = configuration.get(KERBEROS_PRINCIPAL);
+    String keytab = configuration.get(KERBEROS_KEYTAB_FILE);
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java:160]: Need to adapt Linkis's existing kerberos tools?

File: linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/constant/HBaseEngineConnConstant.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+package org.apache.linkis.manager.engineplugin.hbase.constant;
+
+import org.apache.hadoop.hbase.HConstants;
+
+public class HBaseEngineConnConstant {
+  private HBaseEngineConnConstant() {}
+
+  public static final String LINKIS_PREFIX = "wds.linkis.";
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/constant/HBaseEngineConnConstant.java:25]: need to remove wds prefix

File: linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/shell/HBaseShellSessionManager.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      }
+    }
+    return instance;
+  }
+
+  public HBaseShellSession getHBaseShellSession(Map<String, String> prop) {
+    String sessionId = HBaseConnectionManager.getInstance().generateUniqueConnectionKey(prop);
+    LOG.info("Start to create session {} for cluster.", sessionId);
+    try {
+      lock.lock();
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/shell/HBaseShellSessionManager.java:55]: Same as connection and can be optimized

File: linkis-engineconn-plugins/hbase/src/test/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  public void testGetConnection() throws IOException {
+    Map<String, String> properties = new HashMap<>(8);
+    properties.put(HBaseEngineConnConstant.ZOOKEEPER_QUORUM, "localhost");
+    properties.put(HBaseEngineConnConstant.ZOOKEEPER_CLIENT_PORT, "2181");
+    properties.put(HBaseEngineConnConstant.HBASE_SECURITY_AUTH, "simple");
+    // properties.put(HBaseEngineConnConstant.HBASE_SECURITY_AUTH, "kerberos");
+    properties.put(HBaseEngineConnConstant.KERBEROS_PRINCIPAL, "hdfs@HADOOP.COM");
+    properties.put(HBaseEngineConnConstant.KERBEROS_KEYTAB_FILE, "/etc/hbase/conf/hdfs.keytab");
+    // 设置kerberos代理用户
+    properties.put(HBaseEngineConnConstant.KERBEROS_PROXY_USER, "leojie");
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/src/test/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManagerTest.java:47]: leojie should rename to hadoop

File: linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java (hunk 2 above)
Comment (CCweixiao) [reply linkis-engineconn-plugins/hbase/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java]: I feel it is necessary to optimize it later.

File: linkis-engineconn-plugins/hbase/hbase-core/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+        new Thread(
+            new Runnable() {
+              @Override
+              public void run() {
+                while (true) {
+                  int times = 0;
+
+                  while (times < KERBEROS_RE_LOGIN_MAX_RETRY) {
+                    if (runKerberosLogin()) {
+                      LOG.info("Ran kerberos re login command successfully.");
```
Comment (casionone) [inline linkis-engineconn-plugins/hbase/hbase-core/src/main/java/org/apache/linkis/manager/engineplugin/hbase/HBaseConnectionManager.java:222]: Ran ?

File: linkis-dist/release-docs/licenses/LICENSE-jaxb-impl.txt [hunk 7]
```diff
@@ -0,0 +1,759 @@
+COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (CDDL) Version 1.1
```
Comment (peacewong) [inline linkis-dist/release-docs/licenses/LICENSE-jaxb-impl.txt:1]: Is there a GPL LICENSE for this?

File: linkis-dist/release-docs/licenses/LICENSE-jersey-guice.txt [hunk 8]
```diff
@@ -0,0 +1,759 @@
+COMMON DEVELOPMENT AND DISTRIBUTION LICENSE (CDDL) Version 1.1
+
```
Comment (peacewong) [inline linkis-dist/release-docs/licenses/LICENSE-jersey-guice.txt:2]: Is there a GPL LICENSE for this?

File: linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/conf/HBaseConfiguration.scala [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.manager.engineplugin.hbase.conf
+
+import org.apache.linkis.common.conf.CommonVars
+
+object HBaseConfiguration {
+  val HBASE_CONCURRENT_LIMIT = CommonVars[Int]("wds.linkis.engineconn.hbase.concurrent.limit", 100)
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/conf/HBaseConfiguration.scala:23]: remove wds prefix

File: linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+      shellSession = shellSessionManager.getHBaseShellSession(properties)
+    }) { e: Throwable =>
+      logger.error(s"created hbase shell session error! $e")
+      return ErrorExecuteResponse("created hbase shell session error!", e)
+    }
+    val result: Result = shellSession.execute(realCode)
+    if (!result.isSuccess) {
+      return ErrorExecuteResponse(result.getResult, result.getThrowable)
+    }
+    engineExecutorContext.appendStdout(result.getResult)
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala:95]: Is it necessary to add a result set？

File: linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+
+  override def killAll(): Unit = {
+    logger.info("Killing all query task.")
+
+    logger.info("All query task has killed successfully.")
+  }
+
+  override def killTask(taskId: String): Unit = {
+    logger.info(s"Killing hbase query task $taskId")
+    super.killTask(taskId)
```
Comment (peacewong) [inline linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala:178]: need to support kill task

Comment (CCweixiao) [reply linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala:178]: Supports kill task, which will destroy the hbase-shell session. Every time you kill the task and rerun the task, the session will be restarted.

File: linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala (hunk 10 above)
Comment (CCweixiao) [reply linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala:95]: The execution result of HBase Shell is a bunch of strings. Does it need to be packaged into a result set?

Comment (peacewong) [reply linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala:95]: If  not table we need to write text result set

File: linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala (hunk 11 above)
Comment (peacewong) [reply linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala:178]: > Supports kill task, which will destroy the hbase-shell session. Every time you kill the task and rerun the task, the session will be restarted.

If the task can only be killed by ending the session, I think it is also necessary.

File: linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala (hunk 10 above)
Comment (CCweixiao) [reply linkis-engineconn-plugins/hbase/hbase-core/src/main/scala/org/apache/linkis/manager/engineplugin/hbase/executor/HBaseEngineConnExecutor.scala:95]: image
ResultSetFactory.TEXT_TYPE?

**Review by peacewong — APPROVED** [review]
LGTM.

**Pull request merged**